test/message_bus/message_outbox/declareMessage.js

```diff
@@ -0,0 +1,107 @@
+
+'use strict';
+
+const BN = require('bn.js');
+
+const { AccountProvider } = require('../../test_lib/utils.js');
+const MessageBusUtils = require('../messagebus_utils.js');
+const Utils = require('../../test_lib/utils.js');
+const web3 = require('../../test_lib/web3.js');
+
+const config = {};
+
+contract('MessageOutbox::declareMessage', (accounts) => {
+  const accountProvider = new AccountProvider(accounts);
+
+  beforeEach(async () => {
+    config.declareMessageArgs = {
+      intentHash: web3.utils.soliditySha3({
+        type: 'bytes32',
+        value: 'intent',
+      }),
+      nonce: new BN(1),
+      feeGasPrice: new BN(5),
+      feeGasLimit: new BN(20),
+      sender: accountProvider.get(),
+      metachainId: Utils.getRandomHash(),
+      inboxAddress: accountProvider.get(),
+      calculatedChannelIdentifier : ''
+    };
+
+    config.outbox = await MessageBusUtils.deployMessageOutbox();
+    config.declareMessageArgs.calculatedChannelIdentifier =
+      await MessageBusUtils.hashChannelIdentifier(
+        config.declareMessageArgs.metachainId,
+        config.outbox.address,
+        config.declareMessageArgs.inboxAddress
+      );
+
+    //for initializing outboundChannelIdentifier
+    await config.outbox.setupMessageOutboxDouble(
+      config.declareMessageArgs.metachainId,
+      config.declareMessageArgs.inboxAddress
+    );
+  });
+
+  contract('Negative Tests', async () => {
+    it('should revert if outbox for message hash is already true', async () => {
+      await config.outbox.declareMessageDouble(
+        config.declareMessageArgs.intentHash,
+        config.declareMessageArgs.nonce,
+        config.declareMessageArgs.feeGasPrice,
+        config.declareMessageArgs.feeGasLimit,
+        config.declareMessageArgs.sender
+      );
+
+      await Utils.expectRevert(
+        config.outbox.declareMessageDouble(
+          config.declareMessageArgs.intentHash,
+          config.declareMessageArgs.nonce,
+          config.declareMessageArgs.feeGasPrice,
+          config.declareMessageArgs.feeGasLimit,
+          config.declareMessageArgs.sender
+        ),
+        'Message already exists in the outbox.',
+      );
+    });
+  });
+
+  contract('Positive Tests', async () => {
+    it('should declare a new message', async () => {
+      const newMessageHash = await config.outbox.declareMessageDouble.call(
+        config.declareMessageArgs.intentHash,
+        config.declareMessageArgs.nonce,
+        config.declareMessageArgs.feeGasPrice,
+        config.declareMessageArgs.feeGasLimit,
+        config.declareMessageArgs.sender
+      );
+
+      const calculatedMessageHash = await MessageBusUtils.hashMessage(
+        config.declareMessageArgs.intentHash,
+        config.declareMessageArgs.nonce,
+        config.declareMessageArgs.feeGasPrice,
+        config.declareMessageArgs.feeGasLimit,
+        config.declareMessageArgs.sender,
+        config.declareMessageArgs.calculatedChannelIdentifier
+      );
+
```

Comment:
Add assertion for validating the outbox mapping.

Reply:
Done.